libsel4camkes/src/virtqueue_common.c

```diff
@@ -55,47 +55,46 @@ struct vq_buf_alloc *init_vq_allocator(void *mem_pool, unsigned len, size_t bloc
     return allocator;
 }
 
-int camkes_virtqueue_driver_init_common(virtqueue_driver_t *driver, volatile void *buffer, size_t buffer_size,
-                                        void (*notify)(void), size_t block_size)
+int camkes_virtqueue_driver_init_common(virtqueue_driver_t *driver, volatile void *buffer, unsigned queue_len,
+                                        size_t buffer_size, void (*notify)(void), size_t block_size)
 {
     /* Don't check for notify, as it can be NULL in some situations */
     if (!driver || !buffer) {
         return EINVAL;
     }
 
     void *avail_ring = (void *) buffer;
-    void *used_ring = avail_ring + sizeof(vq_vring_avail_t);
-    void *desc = used_ring + sizeof(vq_vring_used_t);
+    void *used_ring = avail_ring + sizeof(vq_vring_avail_t) + sizeof(uint16_t) * queue_len;
```

Comment:
For maintainability, would it make sense to define a 
```
typefef struct {
uint16_t idx;    // is this an index or an ID?
} vq_vring_avail_elem_t
```
here instead of the plan uint16, so describe more clearly what this field holds?